src/ansys/speos/core/source.py

```diff
@@ -1248,3 +1253,495 @@ def delete(self) -> SourceSurface:
         # spectrum & source
         super().delete()
         return self
+
+
+class BaseSourceAmbient(BaseSource):
+    """
+    Super Class for ambient sources.
+
+    Parameters
+    ----------
+    project : ansys.speos.core.project.Project
+        Project in which source shall be created.
+    name : str
+        Name of the source.
+    description : str
+        Description of the source.
+        By default, ``""``.
+    metadata : Optional[Mapping[str, str]]
+        Metadata of the feature.
+        By default, ``{}``.
+    source_instance : ansys.api.speos.scene.v2.scene_pb2.Scene.SourceInstance, optional
+        Source instance to provide if the feature does not have to be created from scratch
+        By default, ``None``, means that the feature is created from scratch by default.
+
+    Notes
+    -----
+    This is a Super class, **Do not instantiate this class yourself**
+    """
+
+    def __init__(
+        self,
+        project: project.Project,
+        name: str,
+        description: str = "",
+        metadata: Optional[Mapping[str, str]] = None,
+        source_instance: Optional[ProtoScene.SourceInstance] = None,
+    ) -> None:
+        super().__init__(
+            project=project,
+            name=name,
+            description=description,
+            metadata=metadata,
+            source_instance=source_instance,
+        )
+
+    class AutomaticSun:
+        """Sun type Automatic.
+
+        By default, user's current time and Ansys France is used a time zone.
+
+        Parameters
+        ----------
+        sun: ansys.api.speos.scene.v2.scene_pb2.AutomaticSun
+            Wavelengths range protobuf object to modify.
+        default_values : bool
+            Uses default values when True.
+        stable_ctr : bool
+            Variable to indicate if usage is inside class scope
+
+        Notes
+        -----
+        **Do not instantiate this class yourself**, use set_sun_automatic method available in
+        source classes.
+        """
+
+        def __init__(
+            self,
+            sun: scene_pb2.AutomaticSun,
+            default_values: bool = True,
+            stable_ctr: bool = False,
+        ) -> None:
+            if not stable_ctr:
+                raise RuntimeError(
+                    "BaseSourceAmbient.AutomaticSun class instantiated outside of class scope"
+                )
+            self._sun = sun
+
+            if default_values:
+                now = datetime.datetime.now()
+                self.set_year(now.year).set_month(now.month).set_day(now.day).set_hour(
+                    now.hour
+                ).set_minute(now.minute)
+                self.set_time_zone().set_longitude().set_latitude()
+
+        def set_year(self, year: int) -> BaseSourceAmbient.AutomaticSun:
```

Comment:
may be use datetime for set as well... just ask for date try convert if fail return error would remove lots of problems -> make it a property and you shoudl be able to do something like

AutomaticSun.date.year =2023